lib/builder/builder.js

```diff
@@ -62,8 +62,11 @@ export default class Builder extends Tapable {
   get plugins () {
     return this.options.plugins.map((p, i) => {
       if (typeof p === 'string') p = { src: p }
-      p.src = this.nuxt.resolvePath(p.src)
-      return { src: p.src, ssr: (p.ssr !== false), name: `plugin${i}` }
+      return {
+        src: this.nuxt.resolvePath(p.src),
+        ssr: (p.ssr !== false),
+        name: p.src.replace(/[^a-zA-Z?\d\s:]/g, '')
```

Comment:
Can't we wrap `p.src` into [path.basename](https://nodejs.org/api/path.html#path_path_basename_path_ext) ?